src/course-libraries/OutOfSyncAlert.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useEffect, useMemo } from 'react';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import { Button } from '@openedx/paragon';
+import { Loop } from '@openedx/paragon/icons';
+import _ from 'lodash';
+import AlertMessage from '../generic/alert-message';
+import { useEntityLinksSummaryByDownstreamContext } from './data/apiHooks';
+import messages from './messages';
+
+interface OutOfSyncAlertProps {
+  showAlert: boolean,
+  setShowAlert: React.Dispatch<React.SetStateAction<boolean>>,
+  courseId: string,
+  onDismiss?: () => void;
+  onReview: () => void;
+}
+/*
+Shows an alert when library components used in the current course were updated and the blocks in course can be updated.
+Dismiss or review action is persisted using localStorage to avoid displaying the alert on every refresh.
+*/
+const OutOfSyncAlert: React.FC<OutOfSyncAlertProps> = ({
+  showAlert,
+  setShowAlert,
+  courseId,
+  onDismiss,
+  onReview,
+}) => {
+  const intl = useIntl();
+  const { data, isLoading } = useEntityLinksSummaryByDownstreamContext(courseId);
+  const outOfSyncCount = useMemo(() => _.sumBy(data, (lib) => lib.readyToSyncCount), [data]);
+  const alertKey = `outOfSyncCountAlert-${courseId}`;
+
+  useEffect(() => {
+    if (isLoading) {
+      return;
+    }
+    if (outOfSyncCount === 0) {
+      localStorage.removeItem(alertKey);
+      setShowAlert(false);
+      return;
+    }
+    const dismissedAlert = localStorage.getItem(alertKey);
+    setShowAlert(parseInt(dismissedAlert || '', 10) !== outOfSyncCount);
+  }, [outOfSyncCount, isLoading, data]);
+
+  const dismissAlert = () => {
+    setShowAlert(false);
+    localStorage.setItem(alertKey, String(outOfSyncCount));
+    onDismiss?.();
+  };
+
+  const reviewAlert = () => {
+    dismissAlert();
+    onReview();
+  };
+
+  return (
+    <AlertMessage
+      title={intl.formatMessage(messages.outOfSyncCountAlertTitle, { outOfSyncCount })}
+      dismissible
+      show={showAlert}
+      icon={Loop}
+      variant="info"
+      onClose={dismissAlert}
+      actions={[
+        <Button
+          onClick={reviewAlert}
+        >
+          {intl.formatMessage(messages.outOfSyncCountAlertReviewBtn)}
+        </Button>,
+      ]}
+    />
+  );
+};
+
+export default OutOfSyncAlert;
```

Comment:
Nit: personally I try to avoid default exports - they [create inconsistencies](https://dev.to/phuocng/avoid-using-default-exports-a1c) and also [require this weird syntax to mock with Jest](https://github.com/openedx/frontend-app-authoring/blob/85b57301140deb776ed402a194568411d00ba234/src/CourseAuthoringRoutes.test.jsx#L26). Since [October](https://github.com/openedx/eslint-config/pull/164) they are no longer required by our linter.